File: codes/jeongah/35_searchinsertposition.py
class Solution(object):
    def searchInsert(self, nums, target):
        left = 0
        right = len(nums) - 1

        while left <= right:
            middle = (left + right) // 2
            if target == nums[middle]:
                return middle
            elif target > nums[middle]:
                left = middle + 1
            else:
                right = middle - 1

        return left

File: codes/beomjin/35_searchinsertposition.scala
object Solution {
    def searchInsert(nums: Array[Int], target: Int): Int = {
        val lastMinPosition = nums.lastIndexWhere(_ < target)

        if (lastMinPosition == -1) {
            throw new RuntimeException("no such element")
        }

        lastMinPosition + 1
    }
}